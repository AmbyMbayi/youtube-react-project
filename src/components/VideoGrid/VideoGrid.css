.grid__section {
    padding-top: 10px;
}

.grid__title {
    width: fit-content;
    font-size: 18px;
    margin-left: 60px;
}

.video__grid {
    display: flex;
    flex-wrap: wrap;
    padding-left: 40px;
}

.grid__divider {
    width: 1550px;
    margin-left: 50px;
    height: 1px;
    background-color: rgb(182, 177, 177);
    margin-bottom: 20px;
}