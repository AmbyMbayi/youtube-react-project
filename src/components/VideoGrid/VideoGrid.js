import React from 'react';
import VideoPreview from '../VideoPreview/VideoPreview';
import './VideoGrid.css';

function VideoGrid(props) {
	return (
		<div className="grid__section">
			<p className="grid__title">{props.title}</p>
			<div className="video__grid">
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
				<VideoPreview />
			</div>
			<div className="grid__divider" />
		</div>
	);
}

export default VideoGrid;
