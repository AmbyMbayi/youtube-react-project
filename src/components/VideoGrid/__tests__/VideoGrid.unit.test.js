import React from 'react';
import { shallow } from 'enzyme';
import VideoGrid from '../VideoGrid';

describe('VideoGrid', () => {
	test('renders without props', () => {
		const wrapper = shallow(<VideoGrid />);
		expect(wrapper).toMatchSnapshot();
	});
	test('renders with title props', () => {
		const wrapper = shallow(<VideoGrid title="trending" />);
		expect(wrapper).toMatchSnapshot();
	});
});
