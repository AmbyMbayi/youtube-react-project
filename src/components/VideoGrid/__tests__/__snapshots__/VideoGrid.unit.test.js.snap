// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders with title props 1`] = `
<div
  className="grid__section"
>
  <p
    className="grid__title"
  >
    trending
  </p>
  <div
    className="video__grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <div
    className="grid__divider"
  />
</div>
`;

exports[`VideoGrid renders without props 1`] = `
<div
  className="grid__section"
>
  <p
    className="grid__title"
  />
  <div
    className="video__grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <div
    className="grid__divider"
  />
</div>
`;
